controllers/argocd/openshift/openshift.go

```diff
@@ -92,12 +92,37 @@ func ReconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}, hint string) err
 			}
 			policyRules := getPolicyRuleForApplicationController()
 			policyRules = append(policyRules, clusterRole.Rules...)
+			if !argoExists(policyRules) {
+				policyRules = append(policyRules, getPolicyRuleForNonOlmArgoCDApplicationController()...)
+			}
 			o.Rules = policyRules
 		}
 	}
 	return nil
 }
 
+// checks if argoproj.io rules exist in application controller role, if not return false
+func argoExists(policies []rbacv1.PolicyRule) (result bool) {
+	result = false
+	for _, policy := range policies {
+		if contains(policy.APIGroups, "argoproj.io") {
+			result = true
+			break
+		}
+	}
+	return result
+}
+
+// utility function to check for string in a slice
+func contains(s []string, e string) bool {
```

Comment:
Pick better names. Having code that reads like ` a == e` is not ideal. For example use `s` for string and `slice` for the slice. Or something like this. Same thing for the name of the function, `contains` is very generic. Some ideas: `stringExistsInSlice`